h2o-genmodel/src/main/java/hex/genmodel/MojoPipelineBuilder.java

```diff
@@ -0,0 +1,93 @@
+package hex.genmodel;
+
+import hex.genmodel.utils.IOUtils;
+
+import java.io.*;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+import java.util.zip.ZipOutputStream;
+
+public class MojoPipelineBuilder {
+
+  private final Map<String, File> _files = new HashMap<>();
+  private final Map<String, MojoModel> _models = new HashMap<>();
+  private final Map<String, String> _mapping = new HashMap<>();
+  private String _mainModelAlias;
+
+  public MojoPipelineBuilder addModel(String modelAlias, File mojoFile) throws IOException {
+    MojoModel model = MojoModel.load(mojoFile.getAbsolutePath());
+    _files.put(modelAlias, mojoFile);
+    _models.put(modelAlias, model);
+    return this;
+  }
+
+  public MojoPipelineBuilder addMainModel(String modelAlias, File mojoFile) throws IOException {
+    addModel(modelAlias, mojoFile);
+    _mainModelAlias = modelAlias;
+    return this;
+  }
+
+  public MojoPipelineBuilder addMappings(List<MappingSpec> specs) {
+    for (MappingSpec spec : specs) {
+      addMapping(spec);
+    }
+    return this;
+  }
+
+  public MojoPipelineBuilder addMapping(MappingSpec spec) {
+    return addMapping(spec._columnName, spec._modelAlias, spec._predsIndex);
+  }
+
+  public MojoPipelineBuilder addMapping(String columnName, String sourceModelAlias, int sourceModelPredictionIndex) {
+    _mapping.put(columnName, sourceModelAlias + ":" + sourceModelPredictionIndex);
+    return this;
+  }
+
+  public void buildPipeline(File pipelineFile) throws IOException {
+    MojoPipelineWriter w = new MojoPipelineWriter(_models, _mapping, _mainModelAlias);
+
+    try (FileOutputStream fos = new FileOutputStream(pipelineFile);
+         ZipOutputStream zos = new ZipOutputStream(fos)) {
+      w.writeTo(zos);
+      for (Map.Entry<String, File> mojoFile : _files.entrySet()) {
+        ZipFile zf = new ZipFile(mojoFile.getValue());
```

Comment:
We should probably `zf.close()` the zipfile as well. Maybe in another try-with-resources block ? Unlikely to leak resources anyway.

Reply:
you are absolutely right! thank you for catching that